pkg/egressgateway/manager.go

```diff
@@ -39,21 +45,40 @@ type Manager struct {
 
 	// epDataStore stores endpointId to endpoint metadata mapping
 	epDataStore map[endpointID]*endpointMetadata
+
+	identityAllocator identityCache.IdentityAllocator
 }
 
 // NewEgressGatewayManager returns a new Egress Gateway Manager.
-func NewEgressGatewayManager(k8sCacheSyncedChecker k8sCacheSyncedChecker) *Manager {
+func NewEgressGatewayManager(k8sCacheSyncedChecker k8sCacheSyncedChecker, identityAlocator identityCache.IdentityAllocator) *Manager {
 	manager := &Manager{
 		k8sCacheSyncedChecker: k8sCacheSyncedChecker,
 		policyConfigs:         make(map[policyID]*PolicyConfig),
 		epDataStore:           make(map[endpointID]*endpointMetadata),
+		identityAllocator:     identityAlocator,
 	}
 
 	manager.runReconciliationAfterK8sSync()
 
 	return manager
 }
 
+// getIdentityLabels waits for the global identities to be populated to the cache,
+// then looks up identity by ID from the cached identity allocator and return its labels.
+func (manager *Manager) getIdentityLabels(securityIdentity uint32) (labels.Labels, error) {
+	identityCtx, cancel := context.WithTimeout(context.Background(), option.Config.KVstoreConnectivityTimeout)
+	defer cancel()
+	if err := manager.identityAllocator.WaitForInitialGlobalIdentities(identityCtx); err != nil {
```

Comment:
This looks like an initialization step, should it be moved in `NewEressGatewayManager()`?

Reply:
As Gilberto suggested, or use `atomic.once`. Make sure it's not called every time `getIdentityLabels` is called.

Reply:
@jibi this is called in the EndpointUpdate callback, so no good place to relocate the wait. It only needs to be done for the very first time. So I wrapped it in a sync.once.

Reply:
This can remain as is. Resolving.